### PR TITLE
Menu APIs Migration

### PR DESCRIPTION
**Pull request by abhijithcs**

### CODE REVIEW

File: resources/acceleron-server/controllers/MenuController.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      return await this.MenuService.getAllItems(categoryName).catch(error => {
+        throw error
+      });
+    }
+
+  async createNewItem() {
+    const categoryName = this.request.params.categoryName;
+    const itemObject = this.request.body;
+    if(_.isEmpty(categoryName) || _.isNull(itemObject.code) || _.isEmpty(itemObject.name) || _.isEmpty(itemObject.price) ){
+      throw new ErrorResponse(ResponseType.BAD_REQUEST, ErrorType.invalid_data_format)
```
Comment (abhijithcs) [inline resources/acceleron-server/controllers/MenuController.js:99]: Keep specific error messages

File: resources/acceleron-server/controllers/MenuController.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      throw error
+    });
+  }
+
+  async updateItemByCode() {
+    const categoryName = this.request.params.categoryName;
+    const itemCode = this.request.params.itemCode;
+    const newItemObject = this.request.body;
+
+    if(_.isEmpty(categoryName) || _.isEmpty(itemCode) || _.isEmpty(newItemObject.code) || _.isEmpty(newItemObject.name) || _.isEmpty(newItemObject.price)){
```
Comment (abhijithcs) [inline resources/acceleron-server/controllers/MenuController.js:127]: Keep specific error messages - where ever user input is taken

File: resources/acceleron-server/models/MenuModel.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+"use strict";
+let BaseModel = ACCELERONCORE._models.BaseModel;
+
+var _ = require('underscore');
+
+class MenuModel extends BaseModel{
+
+    constructor(request) {
+        super(request);
+        this.couch = ACCELERONCORE._connectors.CouchDB;
```
Comment (abhijithcs) [inline resources/acceleron-server/models/MenuModel.js:10]: Migrate to async Connector - implement Axios module

File: resources/acceleron-server/services/MenuService.js [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+      const data = await this.MenuModel.getMenu().catch(error => {
+        throw error
+      });
+
+      if(_.isEmpty(data)){
+        throw new ErrorResponse(ResponseType.NO_RECORD_FOUND, ErrorType.no_matching_results);
+      }
+      else{
+          if(data.value == undefined){
```
Comment (abhijithcs) [inline resources/acceleron-server/services/MenuService.js:23]: Use loadash module for null / empty checks

File: resources/acceleron-server/services/MenuService.js [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    async getCategoryList() {
+      const data = await this.MenuModel.getCategoryList().catch(error => {
+        throw error
+      });
+
+      if(_.isEmpty(data)){
+        throw new ErrorResponse(ResponseType.NO_RECORD_FOUND, ErrorType.no_matching_results);
+      }
+      else{
+          if(data.value == undefined){
```
Comment (abhijithcs) [inline resources/acceleron-server/services/MenuService.js:40]: Use loadash for empty checks

File: resources/acceleron-server/services/MenuService.js [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+              return data; 
+          }
+      }
+      }
+  
+    async createNewCategory(categoryName) {
+      const categoryData = await this.getCategoryList().catch(error => {
+        throw error
+      });
+      // const menuData = await this.getFullMenu().catch(error => {
```
Comment (abhijithcs) [inline resources/acceleron-server/services/MenuService.js:53]: Remove unwanted comments.

File: resources/acceleron-server/services/MenuService.js [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+      //   throw error
+      // });
+
+        var categoryList = categoryData.value;
+        // var menu = menuData.value;
+
+        for (var i=0; i<categoryList.length; i++) {
+          if (categoryList[i] == categoryName){
+             throw new ErrorResponse(ResponseType.CONFLICT, "Category already exists");
```
Comment (abhijithcs) [inline resources/acceleron-server/services/MenuService.js:62]: Move to the Constant

**Comment by abhijithcs** [comment]
Create a new PR

**Comment by harisk2000** [comment]
Haven't checked the functionality of all, so few basic errors might be there.

**Pull request merged**